src/database_observer.py

```diff
@@ -30,18 +30,19 @@ class DatabaseObserver(Object):
 
     _RELATION_NAME = "database"
 
-    def __init__(self, charm: CharmBase):
+    def __init__(self, charm: CharmBase, relation_name: str = _RELATION_NAME):
```

Comment:
I propose to remove the default (we don't want it to silently fail in a subtle way)

Reply:
Do you have an example of how it could fail, we might want to put that as a test? The default is how we normally use that, all our charms have that constant of `database`, for synapse it also has custom logic to prepare the database.

Reply:
I was thinking of the possibility that a dev forgets to specify the relation_name value while initializing a new DatabaseObserver. That could cause havoc without being immediately noticeable.  
By removing the default, we make sure that such errors cannot happen.

Reply:
The default db is part of the "default" behavior, meaning that if you want to initialize a DatabaseObserver, normally you do it for Synapse and it is the main db, the relation is "database" by convention and that name is set across several charms. The metadata file guards you against relating twice (and I wrote a test to cover that, so if the limit of 1 is raised, it will fail), so it won't fail silently, it will fail loud as it is supposed to and we keep some user ergonomics to point at a DatabaseObserver and it does the job without specifying a name. I'm still torn though as I get your point. @gregory-schiano @jdkandersson @weiiwang01 ?

Reply:
I don't have a strong opinion, would probably lean towards not having the default for safety